### PR TITLE
Remove option of loading from non-unified summary files

### PR DESCRIPTION
**Pull request by oyvindeide**
If we are not able to find the expected unified summary file in the callback we assume it is a non-unified summary file. However that is very unlikely as hardly anyone uses that anymore. It is more likely that the expected summary file is not there. This notifies the user that we are not able to load, and it will be set as failed in the state map.  

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: libres/lib/enkf/forward_load_context.cpp
Comment (BjarneHerland) [inline libres/lib/enkf/forward_load_context.cpp:77]: Return here?

Comment (oyvindeide) [reply libres/lib/enkf/forward_load_context.cpp:77]: Considered it, however there are several conditions that could lead to this:
```
    if (summary)
        load_context->ecl_sum = summary;
    else
        forward_load_context_update_result(load_context, LOAD_FAILURE);
```
which is checked at the end, one of which is the error I am updating here, but also this:
```
if (ecl_sum_get_end_time(summary) < end_time) 
```
so I left it as it was.

**Review by BjarneHerland — CHANGES_REQUESTED** [review]
As discussed - move lines 79-146 into if-block, leave else-block to only report error, and then sum up in the end. Otherwise nice!

**Review by jondequinor — COMMENTED** [review]
There are 1.0e100 functions in ERT, but loading summary data from unified summary files isn't one of them? :face_with_spiral_eyes: 

Looks good! But is is at all possible to detect this earlier than in the forward model?

**Comment by oyvindeide** [comment]
> Looks good! But is is at all possible to detect this earlier than in the forward model?

Probably not at the moment, we should implement more validation on the types and number of files we want produced, but at the moment when we come to collect the files we have an assumption that if it expects summary files to be present it will first try to load unified summary, and if that fails it will assume that unified summary files are produced instead. The latter is quite unlikely as nobody:tm: has been using that for at least a decade.

We should do a larger refactor here, so this is not meant as the end all of solutions, just a minor improvement.

Did that answer your comment?

**Review by jondequinor — APPROVED** [review]
Yes, it answered my question. Thanks!

**Comment by oyvindeide** [comment]
For information: just waiting for confirmation that non-unified summary files are as rare as I claim they are before I merge.

**Review by berland — APPROVED** [review]
Confirmed from leading advisor that it is ok to remove support for non-unified. If Non-unified becomes critical due to simulation time issue, non-unified files can be converted to unified in a forward model.

**Pull request merged**
